AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -210,8 +218,13 @@ public final class APIClient: APIClientProtocol {
     private func handle<R: Request>(
         _ result: URLSessionDownloadSuccess,
         _ request: R
-    ) -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
+    ) throws -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
         log(result: result, request: request)
+
+        if let data = try? Data(contentsOf: result.url) {
+            try responseValidator?.validateReponse(data, for: request, with: result.headers)
```

Comment:
would be nice to unit test this? assert that the validator is called using a mock validator

Reply:
updated, thanks